### PR TITLE
BZ1218308: [GSS](6.1.z) Non existing imports in rdrl files prevents the rule to be opened in Business Central

### PR DESCRIPTION
**Pull request by manstis**
Additional test (and fix) for an edge-case reported by @yurloc 

Furthermore (all be it a hack for possible other bugs) if there's any Exception thrown during unmarshalling we fall-back to using a FreeFormLine to hold the DRL (so at least the user can open and edit the rule).

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java [hunk 1]
```diff
@@ -3651,9 +3657,15 @@ private ExpressionFormLine parseExpression( final RuleModel m,
...
-                                                                   currentFact.getType() ) );
+                    //If the bound type is not in the DMO it probably hasn't been imported.
+                    //So we have little option than to fall back to keeping the value as Text.
+                    if ( currentFact == null ) {
+                        expression.appendPart( new ExpressionText( expressionPart ) );
+                    } else {
+                        expression.appendPart( new ExpressionVariable( expressionPart,
+                                                                       currentFact.getClassName(),
+                                                                       currentFact.getType() ) );
+                    }
```
Comment (Rikkola) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/RuleModelDRLPersistenceImpl.java:3668]: Something I like to do. When all the options for the if do the same thing, in this case they append a part, I usually like to extract a method. This way when reading the code afterwards the methods stay shorter and are faster to read.

``` java
expression.appendPart( getPart(currentFact,expressionPart) );

private ExpressionPart getPart(Something currentFact,ExpressionPart expressionPart){
  // now the if goes here
  if(currentFact == null){
    return new ExpressionText( expressionPart );
  } else {
    return new ExpressionVariable( expressionPart,
      + currentFact.getClassName(),
      + currentFact.getType() );
}
```

**Comment by Rikkola** [comment]
https://github.com/droolsjbpm/drools/commit/9c371d95d
